File: EESoft/DataCenter/DataCenter - Client/Forms/TransferFrm.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using System.IO;
using System.Net.Sockets;
using System.Text;
using FunctionLib;

namespace DataCenter_Client
{
	public partial class TransferFrm
	{
        public string FileID;
        public string FilePath;
		public TransferFrm()
		{
			InitializeComponent();
		}
        public void BtnBrowse_Click(Object sender, EventArgs e)
        {
            OpenFileDialog OFD = new OpenFileDialog();
            FileInfo FileInf;
            OFD.Filter = "All Files (*.*)|*.*";
            OFD.CheckFileExists = true;
            OFD.CheckPathExists = true;
            if (OFD.ShowDialog() == DialogResult.OK)
            {
                FileInf = new FileInfo(OFD.FileName);
                if ((int)(FileInf.Length / 1048576) > 50)
                    MCls.msg("الملف اكبر من 50 ميجا", true, "", this);
                else
                {
                    LblPath.Text = OFD.FileName;
                    BtnUpload.Enabled = true;
                }
            }
            else
            {
                LblPath.Text = "من فضلك اختار مسار الملف لارساله";
                BtnUpload.Enabled = false;
            }
        }
        public void BtnDownload_Click(Object sender, EventArgs e)
        {
            OpenFileDialog OFD = new OpenFileDialog();
            OFD.Filter = "All Files (*.*)|*.*";
            OFD.CheckFileExists = false;
            OFD.CheckPathExists = true;
            if (OFD.ShowDialog() == DialogResult.OK)
                ReciveFile(MainModule.ServerIP, int.Parse(MainModule.REQPort), MainModule.ProjectsPath + FilePath, OFD.FileName);
        }
        public void ReciveFile(string IP_Address, int RPort, string ServerFilePath, string ClientFilePath)
        {
            Enabled = false;
            Socket Sok = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
            short Readed;
            FileStream FileSTM;
            try
            {
                string FullString = (int)NetworkFunction.Commands.ReciveFile + ServerFilePath;
                byte[] DataB = new byte[4097];
                DataB = Encoding.Default.GetBytes(FullString);
                Sok.Connect(IP_Address, RPort);
                if (Sok.Connected == true)
                {
                    Sok.Send(DataB, DataB.Length, SocketFlags.None); //Sending Header
                    DataB = new byte[4097];
                    FileSTM = File.Create(ClientFilePath.Trim());
                    do
                    {
                        Readed = (short)(Sok.Receive(DataB, 0, 4096, SocketFlags.None));
                        if (Readed > 0)
                        {
                            FileSTM.Write(DataB, 0, Readed);
                            LblBytes.Text = Convert.ToString((Conversion.Val(LblBytes.Text.Trim()) + Readed));
                        }
                        else
                            break;
                        Application.DoEvents();
                    } while (true);
                    FileSTM.Close();
                    LblBytes.Text = "0";
                    MCls.msg("تم التحميل", false, "", this);
                }
                else
                {
                    MCls.msg("لا يوجد اتصال", true, "", this);
                }
                Sok.Close();
            }
            catch (Exception ex)
            {
                MCls.msg(ex.Message, true, "", this);
            }
            Enabled = true;
            Sok.Dispose();
            Sok = null;
        }
        public void BtnUpload_Click(Object sender, EventArgs e)
        {
            if (MessageBox.Show("هل انت متأكد؟", "تحزير", MessageBoxButtons.YesNo, MessageBoxIcon.Exclamation) == DialogResult.No)
                return;
            Enabled = false;
            PBUpload.Visible = true;
            string FullString;
            byte[] Buffers = new byte[4097];
            Socket Sok = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
            int Created;
            try
            {
                FileStream FileSRM = new FileStream(LblPath.Text, FileMode.Open, FileAccess.Read);
                if (FileSRM.Length == 0)
                {
                    MCls.msg("حجم الملف 0", true, "", this);
                    return;
                }
                FullString = (int)NetworkFunction.Commands.SendFile + MainModule.ProjectsPath + FilePath + "/" + MainModule.DeletedFilesPath + DateTime.Now.Year + "-" + DateTime.Now.Month + "-" + DateTime.Now.Day + "\\" + "/" + FileSRM.Length;
                PBUpload.Maximum = (int)(FileSystem.FileLen(LblPath.Text));
                PBUpload.Value = 0;
                Buffers = Encoding.Default.GetBytes(FullString);
                Sok.Connect(MainModule.ServerIP, int.Parse(MainModule.REQPort));
                if (Sok.Connected == true)
                {
                    Sok.Send(Buffers, FullString.Length, SocketFlags.None);
                    Sok.Receive(Buffers, 1, SocketFlags.None);
                    Created = Convert.ToInt32(Conversion.Val(Buffers[0]));
                    if (Created == 1)
                    {
                        Buffers = new byte[4097];
                        do
                        {
                            Created = FileSRM.Read(Buffers, 0, 4096);
                            if (Created > 0)
                            {
                                Sok.Send(Buffers, Created, SocketFlags.None);
                                MainModule.FrmTransfer.PBUpload.Value += Created;
                                MainModule.FrmTransfer.Text = "   تحميل الملف  " + Convert.ToInt32((MainModule.FrmTransfer.PBUpload.Value / MainModule.FrmTransfer.PBUpload.Maximum) * 100) + " %";
                            }
                            else
                            {
                                break;
                            }
                            Application.DoEvents();
                        } while (true);
                        MainModule.FrmTransfer.Text = "خصائص الملف الجديد";
                        MainModule.FrmTransfer.PBUpload.Value = 0;
                        MainModule.FrmTransfer.PBUpload.Maximum = 100;
                        MCls.msg("تم التحميل", false, "", this);
                    }
                    else
                    {
                        Sok.Close();
                        Sok = null;
                        MCls.msg("لم يتم تحميل الملف", true, "", this);
                    }
                }
            }
            catch (Exception ex)
            {
                MCls.msg(ex.Message, true, "", this);
            }
            Enabled = true;
            PBUpload.Visible = false;
        }
	}
}

File: EESoft/DataCenterVB/Backup/DataCenter - Client/Forms/TransferFrm.vb
Imports System.IO
Imports System.Net
Imports System.Net.Sockets
Imports System.Text
Public Class TransferFrm
    Private Int1 As Integer
    Private Sub BtnBrowse_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnBrowse.Click
        Dim OFD As New OpenFileDialog

        Dim FileInf As System.IO.FileInfo
        OFD.Filter = "All Files (*.*)|*.*"
        OFD.CheckFileExists = True
        OFD.CheckPathExists = True
        If OFD.ShowDialog = Windows.Forms.DialogResult.OK Then
            FileInf = New System.IO.FileInfo(OFD.FileName)
            If CInt(FileInf.Length / 1048576) > 50 Then
                MsgBox_("File must be less than 50M", 94, 23, False, Me)
            Else
                LblPath.Text = OFD.FileName
                BtnUpload.Enabled = True
            End If
        Else
            LblPath.Text = "Please select the file path to upload it"
            BtnUpload.Enabled = False
        End If
    End Sub
    Private Sub BtnDownload_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnDownload.Click
        Dim OFD As New OpenFileDialog
        OFD.Filter = "All Files (*.*)|*.*"
        OFD.CheckFileExists = False
        OFD.CheckPathExists = True
        If OFD.ShowDialog = Windows.Forms.DialogResult.OK Then
            ReciveFile(ServerIP, REQPort, AllFoldersWithPrv.FolderInfo(FrmMainWindow.TVFolders.SelectedNode.Tag).Files(FrmMainWindow.LVFiles.SelectedIndices(0)).FilePath, OFD.FileName)
        End If
    End Sub
    Public Sub ReciveFile(ByVal IP_Address As String, ByVal RPort As Int32, ByVal ServerFilePath As String, ByVal ClientFilePath As String)
        Me.Enabled = False
        Dim Sok As New Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp)
        Dim Readed As Int16
        Dim FileSTM As FileStream
        Try
            Dim FullString As String = FunctionLib.NetworkFunction.Commands.ReciveFile & ServerFilePath
            Dim DataB(4096) As Byte
            DataB = Encoding.Default.GetBytes(FullString)
            Sok.Connect(IP_Address, RPort)
            If Sok.Connected = True Then
                Sok.Send(DataB, DataB.Length, SocketFlags.None) 'Sending Header
                ReDim DataB(4096)
                FileSTM = File.Create(ClientFilePath.Trim)
                Do
                    Readed = Sok.Receive(DataB, 0, 4096, SocketFlags.None)
                    If Readed > 0 Then
                        FileSTM.Write(DataB, 0, Readed)
                        LblBytes.Text = Val(LblBytes.Text.Trim) + Readed
                    Else
                        Exit Do
                    End If
                    Application.DoEvents()
                Loop
                FileSTM.Close()
                LblBytes.Text = "0"
                MsgBox_("Downloading complete", 109, 23, True, Me)
            Else
                MsgBox_("No Connection available", 105, 23, False, Me)
            End If
            Sok.Close()
            Sok = Nothing
        Catch ex As Exception
            Sok = Nothing
            MsgBox_(ex.Message, 0, 23, False, Me)
        End Try
        Me.Enabled = True
    End Sub
    Private Sub BtnUpload_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnUpload.Click
        If MsgBox("Are you sure?", MsgBoxStyle.YesNo) = MsgBoxResult.No Then
            Exit Sub
        End If
        Me.Enabled = False
        PBUpload.Visible = True
        Dim FullString As String
        Dim Buffers(4096) As Byte
        Dim Sok As New Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp)
        Dim Created As Integer
        Try
            Dim FileSRM As New FileStream(LblPath.Text, FileMode.Open, FileAccess.Read)
            If FileSRM.Length = 0 Then
                MsgBox_("The file is 0 byte", 127, 23, False, Me)
                Exit Sub
            End If
            FullString = FunctionLib.NetworkFunction.Commands.SendFile & AllFoldersWithPrv.FolderInfo(FrmMainWindow.TVFolders.SelectedNode.Tag).Files(FrmMainWindow.LVFiles.SelectedIndices(0)).FilePath & "/" & DeletedFilesPath & Now.Year & "-" & Now.Month & "-" & Now.Day & "\" & "/" & FileSRM.Length
            PBUpload.Maximum = FileLen(LblPath.Text)
            PBUpload.Value = 0
            Buffers = Encoding.Default.GetBytes(FullString)
            Sok.Connect(ServerIP, REQPort)
            If Sok.Connected = True Then
                Sok.Send(Buffers, FullString.Length, SocketFlags.None)
                Sok.Receive(Buffers, 1, SocketFlags.None)
                Created = Val(Buffers(0))
                If Created = 1 Then
                    ReDim Buffers(4096)
                    Do
                        Created = FileSRM.Read(Buffers, 0, 4096)
                        If Created > 0 Then
                            Sok.Send(Buffers, Created, SocketFlags.None)
                            FrmTransfer.PBUpload.Value += Created
                            FrmTransfer.Text = "   Transfer File  " & CInt((FrmTransfer.PBUpload.Value / FrmTransfer.PBUpload.Maximum) * 100) & " %"
                        Else
                            Exit Do
                        End If
                        Application.DoEvents()
                    Loop
                    FrmTransfer.Text = "New File Properties"
                    FrmTransfer.PBUpload.Value = 0
                    FrmTransfer.PBUpload.Maximum = 100
                    MsgBox_("Transfer complete", 121, 23, True, Me)
                Else : Sok.Close() : Sok = Nothing : MsgBox_("Can't create file or sever is out of space", 51, 23, False, Me)
                End If
            End If
        Catch ex As Exception
            MsgBox_(ex.Message, 0, 23, False, Me)
        End Try
        Me.Enabled = True
        PBUpload.Visible = False
    End Sub
End Class